tensorzero_internal/src/clickhouse.rs

```diff
@@ -98,11 +118,10 @@ impl ClickHouseConnectionInfo {
             Self::Mock { mock_data, .. } => {
                 write_mock(rows, table, &mut mock_data.write().await).await
             }
-            Self::Production {
-                database_url,
-                client,
-                ..
-            } => write_production(database_url, client, rows, table).await,
+            Self::Production { client, .. } => {
+                let database_url = self.get_url()?;
+                write_production(&database_url, client, rows, table).await
```

Comment:
Let's keep things as a secret until the last possible second, i.e. pass the secret into write_production, and get the URL out later.